metrics-sender/src/main/java/io/knotx/metrics/graphite/LowNetworkConsumptionGraphiteReporter.java

```diff
@@ -0,0 +1,278 @@
+                  graphite,
+                  clock,
+                  prefix,
+                  rateUnit,
+                  durationUnit,
+                  filter,
+                  executor,
+                  shutdownExecutorOnStop,
+                  disabledMetricAttributes);
+      }
+  }
+
+  private Map<String, Object> lastGauges;
+  private Map<String, Long> lastCounters;
+  private Map<String, Long> lastHistograms;
+  private Map<String, Long> lastMeters;
+  private Map<String, Long> lastTimers;
+
+  /**
+   * Creates a new {@link LowNetworkConsumptionGraphiteReporter} instance.
+   *
+   * @param registry               the {@link MetricRegistry} containing the metrics this
+   *                               reporter will report
+   * @param graphite               the {@link GraphiteSender} which is responsible for sending metrics to a Carbon server
+   *                               via a transport protocol
+   * @param clock                  the instance of the time. Use {@link Clock#defaultClock()} for the default
+   * @param prefix                 the prefix of all metric names (may be null)
+   * @param rateUnit               the time unit of in which rates will be converted
+   * @param durationUnit           the time unit of in which durations will be converted
+   * @param filter                 the filter for which metrics to report
+   * @param executor               the executor to use while scheduling reporting of metrics (may be null).
+   * @param shutdownExecutorOnStop if true, then executor will be stopped in same time with this reporter
+   */
+  private LowNetworkConsumptionGraphiteReporter(MetricRegistry registry, GraphiteSender graphite, Clock clock,
+      String prefix, TimeUnit rateUnit, TimeUnit durationUnit, MetricFilter filter,
+      ScheduledExecutorService executor, boolean shutdownExecutorOnStop,
+      Set<MetricAttribute> disabledMetricAttributes) {
+    super(registry, graphite, clock, prefix, rateUnit, durationUnit, filter, executor,
+        shutdownExecutorOnStop, disabledMetricAttributes);
+    lastGauges = new HashMap<>();
+    lastCounters = new HashMap<>();
+    lastHistograms = new HashMap<>();
+    lastMeters = new HashMap<>();
+    lastTimers = new HashMap<>();
+  }
+
+  @Override
+  public synchronized void report(@SuppressWarnings("rawtypes") SortedMap<String, Gauge> gauges,
+      SortedMap<String, Counter> counters, SortedMap<String, Histogram> histograms,
+      SortedMap<String, Meter> meters, SortedMap<String, Timer> timers) {
+    super.report(
+        removeRepeating(gauges, lastGauges, Gauge::getValue),
+        removeRepeating(counters, lastCounters, Counter::getCount),
+        removeRepeating(histograms, lastHistograms, Histogram::getCount),
+        removeRepeating(meters, lastMeters, Meter::getCount),
+        removeRepeating(timers, lastTimers, Timer::getCount));
+  }
+
+  protected static <M extends Metric, V> SortedMap<String, M> removeRepeating(
```

Comment:
I believe the premise that we can just ignore a metric if it keeps coming up with the same name and value multiple times is flawed. This is just how some of the values we want to measure will and should behave.

Reply:
https://github.com/Knotx/knotx-dashboard/pull/4#discussion_r242488525